Assets/_BForBoss/_Core/Scripts/Environment/HealingItemBehaviour.cs

```diff
@@ -6,6 +6,14 @@ namespace Perigon.Entities
     public class HealingItemBehaviour : ItemPickupBehaviour
     {
         [SerializeField] private float _healAmount = 50f;
+        private bool _shouldResetOnlyWhenWaveIncrements = false;
+        private int _currentWaveNumber;
```

Comment:
The issue with putting this logic into this Item, is that this item is now explicitly tied to the context of waves. 

The concept of waves shouldn't be known by the items. But The waveWorldManager can know about it.  If we were to move this to another game. E.g. Mario with healing items, this wouldn't make sense.

Reply:
So do you think like a `ItemTimer` Monobehaviour Component that we attach as a dependancy on `HealingItemBehavioiur` (i.e. [SerializeField] private ItemTimer _itemTimer) and then call on its method when needed?

Something like

`protected override void Update() {_itemTimer.OnUpdate}`? Then at least the item has no direct dependency on the Wave system.

Reply:
Yeah, i think you would need to separate the timing aspect into an `ItemRespawnTimer` class. 
`ItemTimer` is too vague as to what you're referring to.

However, you wouldn't need to call `_itemTimer.onUpdate` and let it be handled automatically, since it's not really related to the pick up actions/logic.